File: c++/Count the Digits That Divide a Number.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int countDigits(int num) {
        int count = 0;
        int originalNum = num;
        
        while (num > 0) {
            int digit = num % 10;
            num /= 10;
            
            // Avoid division by zero
            if (digit != 0 && originalNum % digit == 0) {
                count++;
            }
        }
        
        return count;
    }
};
// Test cases
int main() {
    Solution sol;
    cout << sol.countDigits(7) << endl;   // Output: 1
    cout << sol.countDigits(121) << endl; // Output: 2
    cout << sol.countDigits(1248) << endl; // Output: 4
    return 0;
}

File: python/Count the Digits That Divide a Number.py
class Solution:
    def countDigits(self, num: int) -> int:
        count = 0
        original_num = num
        
        while num > 0:
            digit = num % 10
            num //= 10
            
            # Avoid division by zero
            if digit != 0 and original_num % digit == 0:
                count += 1
        
        return count

# Test cases
if __name__ == "__main__":
    solution = Solution()
    print(solution.countDigits(7))   # Output: 1
    print(solution.countDigits(121)) # Output: 2
    print(solution.countDigits(1248)) # Output: 4

